packages/core/tests/securityKeySignUp.test.ts

```diff
@@ -0,0 +1,190 @@
+import faker from '@faker-js/faker'
+import { RegistrationCredentialJSON } from '@simplewebauthn/typescript-types'
+import { vi } from 'vitest'
+import { interpret, StateFrom } from 'xstate'
+import { waitFor } from 'xstate/lib/waitFor'
+import { AuthMachine, createAuthMachine } from '../src/machines'
+import { BASE_URL } from './helpers/config'
+import {
+  signUpSecurityKeyConflictErrorHandler,
+  signUpSecurityKeyInternalErrorHandler,
+  signUpSecurityKeyNetworkErrorHandler,
+  signUpSecurityKeySuccessHandler,
+  signUpVerifySecurityKeySuccessHandler,
+  signUpVerifySecurityKeySuccessWithSessionHandler
+} from './helpers/handlers'
+import server from './helpers/server'
+import CustomClientStorage from './helpers/storage'
+
+type AuthState = StateFrom<AuthMachine>
+
+const customStorage = new CustomClientStorage(new Map())
+
+const authMachine = createAuthMachine({
+  backendUrl: BASE_URL,
+  clientUrl: 'http://localhost:3000',
+  clientStorage: customStorage,
+  clientStorageType: 'custom',
+  refreshIntervalTime: 1
+})
+
+const mockAuthenticator = () => {
+  vi.mock('@simplewebauthn/browser', () => {
+    return {
+      default: {},
+      startRegistration: vi.fn(async (): Promise<RegistrationCredentialJSON> => {
+        return {
+          id: faker.datatype.uuid(),
+          rawId: faker.datatype.uuid(),
+          response: {
+            clientDataJSON: faker.datatype.string(30),
+            attestationObject: faker.datatype.string(30)
+          },
+          type: 'public-key',
+          clientExtensionResults: {}
+        }
+      })
+    }
+  })
+}
+
+const authService = interpret(authMachine)
+
+beforeAll(() => server.listen({ onUnhandledRequest: 'error' }))
+afterAll(() => server.close())
+
+beforeEach(() => {
+  authService.start()
+})
+
+afterEach(() => {
```

Comment:
Please clean up after each test by restoring the `mockAuthenticator` to its original implementation.